compiler/src/toEC.ml

```diff
@@ -368,34 +371,28 @@ let add_funcs env fds =
     let s = mkname env fd.f_name.fn_name in
     let funs = 
       Mf.add fd.f_name (s, ((*mk_tys*) fd.f_tyout, (*mk_tys*)fd.f_tyin)) env.funs in
-    { env with funs; alls = Ss.add s env.alls } in
+    { env with funs; alls = ref (Ss.add s !(env.alls)) } in
```

Comment:
do you really want to create a new ref or simply update 
env.alls ?

Reply:
We only need to update env.alls, but having a new ref also works.